eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartPaymentFSM.scala

```diff
@@ -66,20 +64,20 @@ class MultiPartPaymentFSM(nodeParams: NodeParams, paymentHash: ByteVector32, tot
     // A sender should not send us additional htlcs for that payment once we've already reached the total amount.
     // However if that happens the only rational choice is to fulfill it, because the pre-image has been released so
     // intermediate nodes will be able to fulfill that htlc anyway. This is a harmless spec violation.
-    case Event(MultiPartHtlc(_, htlc), _) =>
-      require(htlc.paymentHash == paymentHash, s"invalid payment hash (expected $paymentHash, received ${htlc.paymentHash}")
-      log.info(s"received extraneous htlc amountMsat=${htlc.amountMsat}")
-      parent ! ExtraHtlcReceived(paymentHash, PendingPayment(htlc.id, PartialPayment(htlc.amountMsat, htlc.channelId)), None)
+    case Event(part: PaymentPart, _) =>
+      require(part.paymentHash == paymentHash, s"invalid payment hash (expected $paymentHash, received ${part.paymentHash}")
```

Comment:
nit: same as above, this can be migrated to not use string interpolation

Reply:
I can't do that in `require`, only in `log`. And it only works when there's at most 4 arguments so there are two string interpolations that I left.